InstantPrintTool.py

```diff
@@ -44,24 +55,25 @@ def __init__(self, iface, populateCompositionFz=None):
         self.dialogui.comboBox_composers.currentIndexChanged.connect(self.__selectComposer)
         self.exportButton.clicked.connect(self.__export)
         self.helpButton.clicked.connect(self.__help)
-        self.dialogui.buttonBox.button(QDialogButtonBox.Close).clicked.connect(lambda: self.setEnabled(False))
+        self.dialogui.buttonBox.button(QDialogButtonBox.Close).clicked.connect(lambda: self.dialog.hide())
         self.setCursor(Qt.OpenHandCursor)
 
         settings = QSettings()
         if not settings.value("geometry") == None:
             self.dialog.restoreGeometry(settings.value("geometry"))
+            
         
+    def __onDialogHidden(self):
+        self.__cleanup()
+        self.iface.mapCanvas().unsetMapTool(self)
+        QSettings().setValue("geometry", self.dialog.saveGeometry())
+        
+    
     def setEnabled(self, enabled):
```

Comment:
If nobody calls `setEnabled` with enabled=False anymore, you can remove the `enabled` parameter, and perhaps also rename the method to something like `enableTool`.